src/main/kotlin/no/skatteetaten/aurora/boober/model/AuroraDeploymentContext.kt

```diff
@@ -75,68 +78,88 @@ fun List<AuroraDeploymentContext>.createDeployCommand(
     return valid
 }
 
-fun AuroraDeploymentContext.createResources(): Pair<List<ContextErrors>, Set<AuroraResource>?> {
+fun List<AuroraDeploymentContext>.createResources():
+    Map<AuroraDeploymentContext, List<Pair<ContextErrors?, Set<AuroraResource>?>>> {
 
-    val eitherErrorsOrFeatures: List<Pair<ContextErrors?, Set<AuroraResource>?>> = features.map { (feature, adc) ->
+    val generatedSequential = this.associateWith { context -> context.generateResources(RunType.SEQUENTIAL) }
+    val generatedParallel = this.parallelMap { context -> context to context.generateResources(RunType.PARALLEL) }.toMap()
 
+    return this.associateWith { context ->
+        val l1 = generatedSequential[context].orEmpty()
+        val l2 = generatedParallel[context].orEmpty()
+        l1 + l2
+    }
+}
+
+fun AuroraDeploymentContext.generateResources(runType: RunType): List<Pair<ContextErrors?, Set<AuroraResource>?>> =
+    this.features.map { (feature, adc) ->
         val context = this.featureContext[feature] ?: emptyMap()
         try {
-            null to feature.generate(adc, context)
+            if (runType == RunType.SEQUENTIAL) {
+                null to feature.generateSequentially(adc, context)
+            } else {
+                null to feature.generate(adc, context)
+            }
         } catch (e: Throwable) {
             if (e is ExceptionList) {
                 ContextCreationErrors(this.cmd, e.exceptions) to null
             } else {
                 ContextCreationErrors(this.cmd, listOf(e)) to null
             }
         }
+    }.filter {
```

Comment:
Er dette vits? 
eventuelt bør typen ut fra denne funksjonen da ikke inneholde nulltyper?

Reply:
Returverdien kan være  [error og null] eller [null og ressurs] derfor er de nulltyper.
Filtreringen her sørger for at generateResources kun returnerer verdi for features som er konfigurert enten hvor verdi enten er en feil eller ressurs.
Filtreringen gjør det enklere å debugge og sparer Boober for å prosessere en mengde ikke-konfigurerte ressurser.

Fjerner vi filtreringen så vil alle features som er tilgjengelig i boober bli inkludert med [null og emptyList] (dobbelt opp pga sequential og parallel) og prosessert og filtrert ut i flattenErrorsOrFeatures.